lib/pure/nativesockets.nim

```diff
@@ -125,18 +127,26 @@ proc toInt*(p: Protocol): cint
 when not useWinVersion:
   proc toInt(domain: Domain): cshort =
     case domain
+    of AF_UNSPEC:      result = posix.AF_UNSPEC.cshort
     of AF_UNIX:        result = posix.AF_UNIX.cshort
     of AF_INET:        result = posix.AF_INET.cshort
     of AF_INET6:       result = posix.AF_INET6.cshort
-    else: discard
+
+  proc toKnownDomain*(family: cint): Option[Domain] =
```

Comment:
I agree ``Option`` usually is not the way to go but in this case I proposed to use it. An ``unknown`` Domain enum value would mean every ``case`` statement dealing with this would require a branch ``of unknown: doAssert(false, "API usage error")`` and it's nice that ``toKnownDomain`` forces a validity check through the type system.